extensions/iam/oauth2/src/main/java/org/eclipse/dataspaceconnector/iam/oauth2/impl/Oauth2ServiceImpl.java

```diff
@@ -173,43 +177,24 @@ public VerificationResult verifyJwtToken(String token, String audience) {
 
 
     @Nullable
-    private JWSVerifier createVerifier(JWSHeader header, String publicCertificateAlias) {
-        var publicKey = Objects.requireNonNull(configuration.getCertificateResolver().resolveCertificate(publicCertificateAlias)).getPublicKey();
```

Comment:
Hi @paullatzelsperger ,
before I forget, this is the last point that seems weird to me in current implem:
Here for me we should be using the public key of the Authorization server (DAPS instance in case of DAPS). This one should be obtained through the PublicKeyResolver, which is not used at all in the current implem (keys are properly refreshed from the provided URL but they are never used).

Lets discuss that on Monday.